enterprise/lang-js/src/main/java/io/crate/operation/language/JavaScriptUserDefinedFunction.java

```diff
@@ -142,7 +143,7 @@ private static Object processBytesRefInputIfNeeded(Object value) {
         } else if (value instanceof Map) {
             convertBytesRefToStringInMap((Map<String, Object>) value);
         } else if (value instanceof Object[]) {
-            convertBytesRefToStringInList((Object[]) value);
+            value = convertBytesRefToStringInList(value);
```

Comment:
I'd keep the cast on the call-site to keep it close to the `instanceof` check and the `convertBytesRefToStringList` more self-contained. (No implicit contract that the argument must be an object array)
It also avoids the duplicate casts in the `copyOf` call